### PR TITLE
Improve exception logging when we fail to index / transform message

### PR DESCRIPTION
**Pull request by tibrewalpratik17**
- Adds offset value in the log when a record is failed to index / transformed. This will help in faster debugging of problematic events.

- There was a bug where the log was not showing the value of the decodedRow but was giving the address as below:

```
Caught exception while transforming the record: org.apache.pinot.spi.stream.StreamDataDecoderResult@448107c6
```
Updated to show the decoded row value in the log.

### CODE REVIEW

**Review by Jackie-Jiang — APPROVED** [review]
LGTM!

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java [hunk 1]
```diff
@@ -572,6 +572,7 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
       // Decode message
       StreamDataDecoderResult decodedRow = _streamDataDecoder.decode(messagesAndOffsets.getStreamMessage(index));
       msgMetadata = messagesAndOffsets.getStreamMessage(index).getMetadata();
+      StreamPartitionMsgOffset messageOffset = messagesAndOffsets.getNextStreamPartitionMsgOffsetAtIndex(index);
```
Comment (satishd) [inline pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java:575]: It seems `getNextStreamPartitionMsgOffsetAtIndex` returns the offset of the next message for the given index. Should not we use `getMessageOffsetAtIndex(int index)` for getting the current message's offset that we wanted to use in the error messages?

**Review by satishd — COMMENTED** [review]
Thanks @tibrewalpratik17 for the PR, left a comment for the clarification.

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java (hunk 1 above)
Comment (tibrewalpratik17) [reply pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java:575]: Thanks @satishd for pointing this out! Created a follow-up in #12602 and also added logic for tracking filtered events there as discussed offline.

**Pull request merged**